feat(language_server): watch for `fmt.configPath` file content change

Sysix opened the pull request:
The server now sends a request to the client, to watch for `.oxfmtrc.json` or custom configurable `fmt.configPath` file content changes.
The client will send a `workspace/didChangeWatchedFiles` notification to the server.

The client can also send `workspace/didChangeConfiguration`, for now the server will restart both tools.
This will be optimized by looking at which tool is responsible for the file in https://github.com/oxc-project/oxc/pull/14645

Sysix commented on crates/oxc_language_server/src/worker.rs:
```diff
@@ -346,17 +377,53 @@ impl WorkspaceWorker {
+                            .to_owned(),
+                    }),
+                    kind: Some(WatchKind::all()), // created, deleted, changed
+                });
```
Will be fixed in the downstream PR

camc314 commented:
### Merge activity

* **Oct 17, 10:59 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Oct 17, 10:59 AM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).
* **Oct 17, 11:04 AM UTC**: Merged by the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

Sysix committed:
feat(language_server): watch for `fmt.configPath` file content change (#14509)

The server now sends a request to the client, to watch for `.oxfmtrc.json` or custom configurable `fmt.configPath` file content changes.
The client will send a `workspace/didChangeWatchedFiles` notification to the server.

The client can also send `workspace/didChangeConfiguration`, for now the server will restart both tools.
This will be optimized by looking at which tool is responsible for the file in https://github.com/oxc-project/oxc/pull/14645
